BridgeCareApp/VuejsApp/src/components/performance-editor/PerformanceEditor.vue

```diff
@@ -0,0 +1,545 @@
+            this.showCreatePerformanceStrategyEquationDialog = true;
+        }
+
+        /**
+         * User has submitted a createdEquation dialog result
+         */
+        onCreatePerformanceStrategyEquation(createdPerformanceStrategyEquation: CreatedPerformanceStrategyEquation) {
+            // set showCreatePerformanceStrategyEquationDialog to false to hide CreatePerformanceStrategyEquationDialog
+            this.showCreatePerformanceStrategyEquationDialog = false;
+            // if there is a createdPerformanceStrategyEquation
+            if (hasValue(createdPerformanceStrategyEquation)) {
+                // set the createdPerformanceStrategyEquation.performanceStrategyEquationId with selected performance strategy id
+                createdPerformanceStrategyEquation.performanceStrategyId = this.selectedPerformanceStrategy.id;
+                // set isBusy to true, then dispatch an action to create the equation
+                this.setIsBusyAction({isBusy: true});
+                this.createEquationAction({createdEquation: createdPerformanceStrategyEquation})
+                    .then(() => this.setIsBusyAction({isBusy: false}))
+                    .catch((error: any) => {
+                        this.setIsBusyAction({isBusy: false});
+                        console.log(error);
+                    });
+            }
+        }
+
+        /**
+         * 'Toggle Shift' button has been clicked
+         */
+        onToggleShift() {
+            // get the selected equation ids from selectedGridRows
+            const selectedEquationIds: number[] = this.selectedGridRows
+                .map((equation: PerformanceStrategyEquation) => equation.performanceStrategyEquationId);
+            // get the equations to update from selectedPerformanceStrategy.performanceStrategyEquations
+            // and negate their current shift values
+            const equations: PerformanceStrategyEquation[] = this.selectedPerformanceStrategy.performanceStrategyEquations
+                .filter((equation: PerformanceStrategyEquation) => contains(equation.performanceStrategyEquationId, selectedEquationIds))
+                .map((equation: PerformanceStrategyEquation) => ({...equation, shift: !equation.shift}));
+            // dispatch updateEquations action to update the performance strategy equations in state
+            this.updateEquationsAction({updatedEquations: equations});
+        }
+
+        /**
+         * 'Edit Equation' button has been clicked
+         */
+        onShowEquationEditorDialog() {
+        }
+
+        /**
+         * User has submitted PerformanceEquationEditorDialog result
+         * @param equation The submitted equation string
+         */
+        onSubmitEquationEditorResult(equation: string) {
+            // reset equationEditorDialogData
+            this.equationEditorDialogData = {...emptyEquationEditorDialogData};
+            if (!isNil(equation)) {
```

Comment:
can't we use `hasvalue` over here?

Reply:
We can, but then that would disallow users from actually removing a criteria (an empty string). I have a feeling they may want to be able to do that.

Reply:
got it.